backend/sites/src/app/services/document/document.service.ts

```diff
@@ -12,36 +12,56 @@ export class DocumentService {
     private siteDocsRepository: Repository<SiteDocs>,
   ) {}
 
+  /**
+   * Retrieves site documents by site ID and transforms the data into a specific format.
+   * @param siteId - The ID of the site to retrieve documents for.
+   * @returns A promise that resolves to an array of DocumentDto objects.
+   * @throws Error if there is an issue retrieving the data.
+   */
   async getSiteDocumentsBySiteId(siteId: string) {
     try {
-      const result = await this.siteDocsRepository.find({ where: { siteId } });
-      if (result) {
-        const response = result
-          .map((res) => {
-            const document = {
-              id: res.id,
-              siteId: res.siteId,
-              title: res.title,
-              submissionDate: res.submissionDate.toISOString(),
-              documentDate: res.documentDate.toISOString(),
-            };
-            let obj = [{ psnorgId: '', displayName: '' }];
-            if (res.siteDocPartics.length > 0) {
-              obj = res.siteDocPartics.map((sdp) => ({
-                psnorgId: sdp.psnorgId,
-                displayName: sdp.psnorg.displayName,
-              }));
-            }
+      // Fetch documents for the given site ID
+      const result = await this.siteDocsRepository.find({
+        where: { siteId },
+        relations: ['siteDocPartics', 'siteDocPartics.psnorg'], // Fetch related entities
+      });
 
-            const doc = obj.map((item) => ({ ...document, ...item }));
-            return doc;
-          })
-          .flat();
+      // Process and format the documents
+      const response = result.flatMap((res) => {
```

Comment:
please handle in case result is null

Reply:
Result wont be null, if there will be no data in table the it will return empty array, that is handled at resolver level using this if (result && result.length > 0)